Fix error after second start_chat() for StatefulLLMPipeline

sbalandi opened the pull request:
Ticket: [CVS-161902](https://jira.devtools.intel.com/browse/CVS-161902)

ilya-lavrenov commented on src/cpp/src/llm_pipeline_stateful.cpp:
```diff
@@ -373,6 +373,11 @@ EncodedResults StatefulLLMPipeline::generate(
 }
 
 void StatefulLLMPipeline::start_chat(const std::string& system_message) {
+    bool have_state = 0 != m_model_runner.get_tensor("attention_mask").get_size();
```
let's add test

ilya-lavrenov commented on src/cpp/src/llm_pipeline_stateful.cpp:
```diff
@@ -373,6 +373,11 @@ EncodedResults StatefulLLMPipeline::generate(
+    if (have_state) {
+        m_model_runner.reset_state();
+        m_model_runner.get_tensor("attention_mask").set_shape({1, 0});
+    }
```
I think we should also cover a case when user call next `start_chat` w/o `stop_chat` for current one.
IMO, it should automatically stop current chat

sbalandi replied:
cases with system_massage, with double start_chat and with several chats one after another are added. 
If i understand right tests work now for PA backend, should it be added run for SDPA too ?

ilya-lavrenov replied:
> If i understand right tests work now for PA backend, should it be added run for SDPA too ?

they will be added as part of CVS-159925
For now, you can check locally that tests are passing on SPDA branch by changing default backend here https://github.com/openvinotoolkit/openvino.genai/blob/1bdd4f94a4bb610b84f5c28e504c8df7cc332d30/src/cpp/src/llm_pipeline.cpp#L62

ilya-lavrenov commented on src/cpp/src/llm_pipeline_stateful.cpp:
```diff
@@ -373,6 +373,11 @@ EncodedResults StatefulLLMPipeline::generate(
+    if (have_state) {
+        m_model_runner.reset_state();
+        m_model_runner.get_tensor("attention_mask").set_shape({1, 0});
+    }
```
should `start_chat` call `stop_chat` first ?

ilya-lavrenov commented:
does a current PR fix https://github.com/openvinotoolkit/openvino.genai/issues/1663 ?

sbalandi commented:
> does a current PR fix #1663 ?

The reason looks like point which Vladimir [mention above](https://github.com/openvinotoolkit/openvino.genai/pull/1684#discussion_r1945990426). If execution is interrupted and generation() does not complete correctly, this error will occur. I moved the resets to the beginning, this fixes similar behavior for phi-3.

Wovchena commented on tests/python_tests/test_llm_pipeline.py:
```diff
@@ -121,20 +122,24 @@ def test_empty_encoded_inputs_throw():
 
-    ov_pipe.start_chat()
+    ov_pipe.start_chat(system_massage)
+    chat_history_hf.append({"role", "system", "content", system_massage})
+    chat_history_ov.append({"role", "system", "content", system_massage})
```
```suggestion
    chat_history_hf.append({"role": "system", "content": system_massage})
    chat_history_ov.append({"role:, "system", "content": system_massage})
```
You also got conflicts

Wovchena commented:
> I moved the resets to the beginning, this fixes similar behavior for phi-3.

There are other pipelines that need that change. (VLM, at least.) Can you submit a ticket to address it or simply fix in the next PR

sbalandi commented:
> > I moved the resets to the beginning, this fixes similar behavior for phi-3.
> 
> There are other pipelines that need that change. (VLM, at least.) Can you submit a ticket to address it or simply fix in the next PR

added to VLM
